pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert move_df.get_users_number() == 1
+
+    move_df[UID] = [1, 1, 2, 3]
+    assert move_df.get_users_number() == 3
+
+
+def test_to_numpy():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_numpy(), ndarray)
+
+
+def test_to_dict():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_dict(), dict)
+
+
+def test_to_grid():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_grid(8), pymove.core.grid.Grid)
+
+
+def test_to_data_frame():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_data_frame(), DataFrame)
+
+
+def test_describe():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [4.0, 4.0, 4.0],
+        [39.984185, 116.31934049999998, 1.5],
+        [6.189237971348586e-05, 7.921910543639078e-05, 0.5773502691896257],
+        [39.984094, 116.319236, 1.0],
+        [39.984172, 116.3193005, 1.0],
+        [39.984211, 116.319362, 1.5],
+        [39.984224, 116.319402, 2.0],
+        [39.984224, 116.319402, 2.0],
+    ], columns=['lat', 'lon', 'id'],
+        index=['count', 'mean', 'std', 'min', '25%', '50%', '75%', 'max']
+    )
+    assert_frame_equal(move_df.describe(), expected)
+
+
+def test_memory_usage():
+    move_df = _default_move_df()
+
+    expected = Series(
+        data=[128, 32, 32, 32, 32],
+        index=['Index', 'lat', 'lon', 'datetime', 'id'], dtype='int64', name=None
+    )
+    assert_series_equal(move_df.memory_usage(), expected)
+
+
+def test_copy():
+    move_df = _default_move_df()
+
+    cp = move_df.copy()
+    assert_frame_equal(move_df, cp)
+    cp.at[0, TRAJ_ID] = 0
+    assert move_df.loc[0, TRAJ_ID] == 1
+    assert move_df.loc[0, TRAJ_ID] != cp.loc[0, TRAJ_ID]
+
+
+def test_generate_tid_based_on_id_datetime():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_tid_based_on_id_datetime(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, '12008102305'],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, '12008102305'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, '22008102305'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, '22008102305'],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'tid'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)